### DISCUSSION TITLE
Deploy Functions with NO TYPE CHECK Flag on default

### DISCUSSION THREAD

**Discussion by christianbm1** [supabase/supabase#6749] [Feature Requests]
Hello,

I have a function with some type checking issues. The issues are in the libraries that I'm importing, and I couldn't fix them. Locally, using Deno alone, my function runs smoothly when adding the "--no-check" flag. Additionally, my function successfully deploys and works when deploying to Deno's Deploy service. By default, Deno's service automatically adds the no-check flag.

This is not the case for Supabase. Supabase doesn't add this flag for both running locally and on deploy. However, I was able to get around this by manually adjusting the golang Supabase CLI code to force add the no-check flag.

Seeing as Deno Deploy's service doesn't do type check, should the default for Supabase be to add the "no-check" flag for serving locally and on deploy as well? Or, at a minimum, can we have the ability to add this flag at the user's request when running the serve/deploy commands through the sb CLI?

Deno no type checking flag was added to be the default in their v1.21 - can read their blog here: https://deno.com/blog/v1.21

Thanks,
Christian

**Comment by dennismuench** [comment]
Just to give an example:

Stripe is not fully compatible with Deno and cannot be deployed via the Supabase CLI without adding `no-check` inside the `import_map.json` like so: `https://esm.sh/stripe@12.0.0?target=deno&no-check` but this disables typings entirely and casts everything to `any` making it a very poor developing experience. It would of course be much better to leave type checking enabled for development / linting, having the ability to skip checks during deployment. Especially because you advertise Stripe integration in the docs but every user will stumble uppon the same issue eventually.

Reference: https://github.com/supabase/supabase/issues/10543#issuecomment-1452451371
